api/src/adapters/aws/api_gateway_adapter.py

```diff
@@ -0,0 +1,156 @@
+
+from src.adapters.aws import get_boto_session
+from src.adapters.aws.aws_session import is_local_aws
+from src.util.env_config import PydanticBaseEnvConfig
+
+logger = logging.getLogger(__name__)
+
+
+class ApiKeyImportResponse(BaseModel):
+    id: str = Field(alias="id")
+    name: str = Field(alias="name")
+    description: str | None = Field(alias="description", default=None)
+    enabled: bool = Field(alias="enabled", default=True)
+    stage_keys: list[str] = Field(alias="stageKeys", default_factory=list)
+    tags: dict[str, str] = Field(alias="tags", default_factory=dict)
+
+
+class ApiGatewayConfig(PydanticBaseEnvConfig):
+    """Configuration for AWS API Gateway integration"""
+
+    # Usage plan ID for newly created API keys (typically the public usage plan)
+    default_usage_plan_id: str | None = Field(
+        default=None, alias="API_GATEWAY_DEFAULT_USAGE_PLAN_ID"
+    )
+
+
+def get_boto_api_gateway_client(session: boto3.Session | None = None) -> botocore.client.BaseClient:
+    """Get a boto3 API Gateway client"""
+    if session is None:
+        session = get_boto_session()
+
+    return session.client("apigateway")
+
+
+def import_api_key(
+    api_key: str,
+    name: str,
+    description: str | None = None,
+    enabled: bool = True,
+    usage_plan_id: str | None = None,
+    api_gateway_client: botocore.client.BaseClient | None = None,
+) -> ApiKeyImportResponse:
+    """
+    Import an API key into AWS API Gateway using CSV format.
+
+    Args:
+        api_key: The API key value to import
+        name: Name for the API key
+        description: Optional description for the API key
+        enabled: Whether the API key should be enabled (default: True)
+        usage_plan_id: Optional usage plan ID to associate the key with
+        api_gateway_client: Optional pre-configured API Gateway client
+
+    Returns:
+        ApiKeyImportResponse with the imported key details
+    """
+
+    if api_gateway_client is None:
+        api_gateway_client = get_boto_api_gateway_client()
+
+    if is_local_aws():
+        return _handle_mock_import_response(api_key, name, description, enabled, usage_plan_id)
+
+    try:
+        # Format the API key data as CSV for import
+        # AWS API Gateway expects CSV format: name,key,description,enabled
+        csv_data = f"{name},{api_key},{description or ''},{'true' if enabled else 'false'}"
+
+        response = api_gateway_client.import_api_keys(
+            body=csv_data.encode("utf-8"), format="csv", failOnWarnings=True
+        )
+
+        if response.get("warnings"):
+            logger.warning("API Gateway import warnings", extra={"warnings": response["warnings"]})
+
+        imported_key_ids = response.get("ids", [])
+        if not imported_key_ids:
+            raise Exception("No API key IDs returned from import operation")
+
+        key_id = imported_key_ids[0]
+        key_details = api_gateway_client.get_api_key(apiKey=key_id, includeValue=False)
```

Comment:
It shouldn't happen, but should we add a sanity check that it returned exactly 1 key?